packages/react-aria-components/src/Tree.tsx

```diff
@@ -493,7 +509,7 @@ function convertExpanded(expanded: 'all' | Iterable<Key>): 'all' | Set<Key> {
     : new Set(expanded);
 }
 interface TreeGridCollectionOptions {
-  expandedKeys: 'all' | Set<Key>
+  expandedKeys: Set<Key>
```

Comment:
stems from this discussion: https://github.com/adobe/react-spectrum/pull/5756#pullrequestreview-1912741922. May need to create a new hook for useTreeState that supports `all` that would be used by TreeView and Tree but probably would be confusing for users to have two "tree" hooks...